distributed/utils.py

```diff
@@ -1000,3 +1001,42 @@ def __init__(self, *args, **kwargs):
 
     def emit(self, record):
         self.deque.append(record)
+
+
+class ThrottledGC(object):
+    """Wrap gc.collect to protect against excessively repeated calls.
+
+    Allows to run throttled garbage collection in the workers as a
+    countermeasure to e.g.: https://github.com/dask/zict/issues/19
+
+    throttled_gc.collect() does nothing when repeated calls are so costly and
+    so frequent that the thread would spend more than max_in_gc_frac doing GC.
+    """
+    def __init__(self, max_in_gc_frac=0.2, warn_if_longer=10, logger=None):
```

Comment:
Do you have any thoughts on the default value for `max_in_gc_frac=0.2`?  Spending 20% of our time in garbage collection seems unpleasant.

Reply:
The `warn_if_longer=` parameter could also use some mention in the docstring

Reply:
>  Spending 20% of our time in garbage collection seems unpleasant.

It's an upper-bound. In most situations memory-limit is high enough that GC will almost never be called at all.

But in cases where the worker is under heavy memory pressure, I find it better to spend that time in the GC than run the risk to be spuriously killed by nanny for no valid reason. Note that it's not a smoothed average, just based on the last 2 calls to collect.

Reply:
I will try to change the fraction to 5% and run some experiments.

Reply:
In the meantime, I have pushed an improvement to the docstring as requested. Let me know what you think.

Reply:
> It's an upper-bound. In most situations memory-limit is high enough that GC will almost never be called at all.

OK, that makes sense to me. 

> In the meantime, I have pushed an improvement to the docstring as requested. Let me know what you think.

If we're spending 10s in gc then that is probably also something that we should know about.  Thoughts on reducing this to 1s?